src/main/java/com/aws/greengrass/mqttbridge/util/BatchedSubscriber.java

```diff
@@ -48,11 +48,11 @@
             what == WhatHappened.timestampUpdated || what == WhatHappened.interiorAdded;
 
     private final AtomicInteger numRequestedChanges = new AtomicInteger();
-    private final Set<Node> whatChanged = new HashSet<>();
+    private final List<Node> whatChanged = new ArrayList<>();
 
     private final Node node;
     private final BiPredicate<WhatHappened, Node> exclusions;
-    private final BiConsumer<WhatHappened, Set<Node>> callback;
+    private final BiConsumer<WhatHappened, List<Node>> callback;
```

Comment:
A set is more appropriate here

Reply:
Yeah, i only switched it to have a way to get the "last changed node", but per your suggestion to do a lookup, this isn't necessary anymore, will switch it back to set 👍

Reply:
However, I'm more generally worried about storing references to all the nodes, I think that it will encourage bad behavior. I can't really substantiate that, but it is a feeling that it is wrong.